cmd/ipfs/daemon.go

```diff
@@ -181,6 +181,13 @@ func defaultMux(path string) corehttp.ServeOption {
 var fileDescriptorCheck = func() error { return nil }
 
 func daemonFunc(req cmds.Request, res cmds.Response) {
+	// Inject metrics before we do anything
+
+	err := mprome.Inject()
+	if err != nil {
+		log.Warningf("Injecting prometheus handler for metrics failed with message: %s\n", err.Error())
```

Comment:
should this just be a warning? Or should we actually error out here?

Also, can we (in a future PR) have a flag to disable metrics entirely?